Applications/YOLO/PyTorch/yolo_loss.py

```diff
@@ -0,0 +1,205 @@
+    def _build_target(self, bbox_pred, bbox_gt, cls_gt):
+        """
+        @param bbox_pred shape(batch_size, cell_h x cell_w, num_anchors, 4)
+        @param bbox_gt shape(batch_size, num_bbox, 4)
+        @param cls_gt shape(batch_size, num_bbox, 1)
+        @return tuple of (bbox_built, iou_built, cls_built, bbox_mask, iou_mask, cls_mask)
+        """    
+        bbox_built, bbox_mask = [], []
+        iou_built, iou_mask = [], []
+        cls_built, cls_mask = [], []
+        
+        batch_size = bbox_pred.shape[0]
+                
+        for i in range(batch_size):
+            _bbox_built, _iou_built, _cls_built,\
+                _bbox_mask, _iou_mask, _cls_mask =\
+                    self._make_target_per_sample(
+                        torch.FloatTensor(bbox_pred[i]),
+                        torch.FloatTensor(np.array(bbox_gt[i])),
+                        torch.LongTensor(cls_gt[i])
+                    )
+            
+            bbox_built.append(_bbox_built.numpy())
+            bbox_mask.append(_bbox_mask.numpy())
+            iou_built.append(_iou_built.numpy())
+            iou_mask.append(_iou_mask.numpy())
+            cls_built.append(_cls_built.numpy())
+            cls_mask.append(_cls_mask.numpy())
+
+        bbox_built, bbox_mask, iou_built, iou_mask, cls_built, cls_mask =\
+            torch.FloatTensor(np.array(bbox_built)),\
+            torch.FloatTensor(np.array(bbox_mask)),\
+            torch.FloatTensor(np.array(iou_built)),\
+            torch.FloatTensor(np.array(iou_mask)),\
+            torch.FloatTensor(np.array(cls_built)),\
+            torch.FloatTensor(np.array(cls_mask))
+                    
+        return bbox_built, iou_built, cls_built, bbox_mask, iou_mask, cls_mask
+        
+    def _make_target_per_sample(self, _bbox_pred, _bbox_gt, _cls_gt):
+        """
+        @param _bbox_pred shape(cell_h x cell_w, num_anchors, 4)
+        @param _bbox_gt shape(num_bbox, 4)
+        @param _cls_gt shape(num_bbox,)
+        @return tuple of (_bbox_built, _iou_built, _cls_built, _bbox_mask, _iou_mask, _cls_mask)
+        """
+        hw, num_anchors, _  = _bbox_pred.shape
+        
+        # set result template
+        _bbox_built = torch.zeros((hw, num_anchors, 4))
+        _bbox_mask = torch.zeros((hw, num_anchors, 1))
+        
+        _iou_built = torch.zeros((hw, num_anchors, 1))
+        _iou_mask = torch.ones((hw, num_anchors, 1)) * 0.5
+        
+        _cls_built = torch.zeros((hw, num_anchors, self.num_classes))
+        _cls_mask = torch.zeros((hw, num_anchors, 1))
+                        
+        # find best anchors
+        _bbox_gt_wh = _bbox_gt.clone()[:, 2:]        
+        best_anchors = find_best_ratio(self.anchors, _bbox_gt_wh)
+        
+        # normalize x, y pos based on cell coornindates
+        cell_w = float(self.img_shape[0]) / self.outsize[0] 
+        cell_h = float(self.img_shape[1]) / self.outsize[1]
+        cx = _bbox_gt[:, 0] * self.img_shape[0] / cell_w
```

Comment:
How about this?
`cx = _bbox_gt[:, 0] * self.outsize[0]`
I think cell_w, cell_h does not needed

Reply:
> How about this? `cx = _bbox_gt[:, 0] * self.outsize[0]` I think cell_w, cell_h does not needed

That's a good idea. thanks!